SIPS/sip-59.md

```diff
@@ -0,0 +1,141 @@
+---
+sip: 59
+title: Delegated Migrator
+status: Proposed
+author: Justin J Moses (@justinjmoses)
+discussions-to: <https://discordapp.com/invite/AEdUHzt>
+
+created: 2020-05-11
+---
+
+<!--You can leave these HTML comments in your merged SIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new SIPs. Note that an SIP number will be assigned by an editor. When opening a pull request to submit your SIP, please use an abbreviated title in the filename, `sip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SIP.-->
+
+Make upgrades of Synthetix as seamless as possible
+
+## Abstract
+
+<!--A short (~200 word) description of the technical issue being addressed.-->
+
+The introduction of the Delegated Migrator contract to perform upgrades on-chain in one or more steps. Migration scripts, once proposed and after a waiting period has elapsed, can be executed by the ProtocolDAO which will delegate to the migration scdript.
+
+## Motivation
+
+<!--The motivation is critical for SIPs that want to change Synthetix. It should clearly explain why the existing protocol specification is inadequate to address the problem that the SIP solves. SIP submissions without sufficient motivation may be rejected outright.-->
+
+Upgrading the Synthetix protocol currently takes the core contributors a number of hours to run.
+
+A typical upgrade involves replacing the following contract sources: `Synthetix`, `FeePool` and every synth's `Synth` (over 40 at the time of writing), each of which require a change to both their associated proxies (via `setTarget`) and their associated state contracts (via `setAssociatedContract`). There are also changes to populate the `AddressResolver` with the new addresses, and updating each contract that depends on the AddressResolver to sync its cache (thus saving gas for end users).
```

Comment:
It would be better to focus on the Motivation on the path to a decentralised protocol that is censor proof and managed by the token holders as a voting mechanism, discuss the ability to add a time lock to upgrades that can be voted / vetoed  on, (like other protocols),  than discussing the frustration the core contributors have currently as a 3rd party reading the SIP it would be more interest for them to know how this improve decentralised governance.